src/allmydata/storage_client.py

```diff
@@ -54,6 +54,24 @@
 # look like?
 #  don't pass signatures: only pass validated blessed-objects
 
+import os
+
+
+def sort_servers(servers, key):
+    """
+    This is sample server selection hookup function.
+    This is the fallback for no 3rd party server selection configured in tahoe.cfg also.
+    The 3rd party can provide a similar server selection function.
+
+    :param servers: A set of allmydata.storage_client.NativeStorageServer object
+    :param key: Specifies a function of one argument that is used to extract a comparison key from Tahoe server list
+                element. The 3rd party can use this as a default comparison key if they don't want to provide their own
+                comparison algorithm but still want to keep using the server selection framework.
+    :return: A sorted server list
```

Comment:
This seems like a complicated API to export to users -- what they're supposed to do with `key` seems unclear (e.g. what if it's not specified? can it be ignored if it is specified?). I think it would be better if the "sort_servers" function takes only a single argument (the `servers` list). Or, maybe the second argument should be `peer_selection_index`?

Then for this default implementation, you could move the `permuted` inner-method from `get_servers_for_psi` into this function and thus still call `sorted` in the same fashion (with `key=permuted`).